### PR TITLE
Ubuntu download of crun package

### PR DESCRIPTION
**Pull request by cevich**
Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

File: cache_images/ubuntu_packaging.sh
Comment (edsantiago) [inline cache_images/ubuntu_packaging.sh:121]: Why DOWNLOAD_PACKAGES? Why not in INSTALL_PACKAGES?

Comment (cevich) [reply cache_images/ubuntu_packaging.sh:121]: @mheon wanted this so he could test some things.  I didn't want to risk mucking up everybody's tests by having both `runc` and `crun` installed at the same time.  Am I wrong, is that a perfectly safe thing to do?

**Comment by cevich** [comment]
note to me: needs rebase

File: cache_images/ubuntu_packaging.sh
Comment (cevich) [reply cache_images/ubuntu_packaging.sh:121]: more context: None of the Ubuntu VMs have cgv2 enabled by default on the kernel.  @mheon is expecting this.

Comment (edsantiago) [reply cache_images/ubuntu_packaging.sh:121]: I have to assume that podman will figure out whether to use runc or crun; but this has to be decided by @mheon.

@cevich, just so @mheon (and I) can understand the full ramifications, can you please ELI5 the difference between DOWNLOAD and INSTALL? I think I have a small understanding of it but don't want to misspeak.

Comment (mheon) [reply cache_images/ubuntu_packaging.sh:121]: I don't actually know what the default runtime chosen will be on a cgroupsv1 system with runc and crun both installed, I'd need to look at the code to check...

Comment (cevich) [reply cache_images/ubuntu_packaging.sh:121]: Remember too: These VMs are potentially used across multiple projects.  So for example, `buildah` CI would also need to behave with having both installed.  The safest thing is to download it like this, then manage the install/swap at test-time.  This gets consistent/reliable package versions, and won't mess with other automation.  Where were you thinking of testing this from.  Manually or in a script or in ginkgo or somewhere else?

Comment (mheon) [reply cache_images/ubuntu_packaging.sh:121]: I have confirmed that `containers.conf` will correctly select `runc` as the default on a cgroups v1 system if both runc and crun are available. Buildah does *not* use this bit of containers.conf (which seems like an unfortunate oversight...) but it, too, correctly uses runc on v1 systems even if both are installed.

I think it ought to be safe to have both runtimes installed given this.

Comment (cevich) [reply cache_images/ubuntu_packaging.sh:121]: Thanks guys.  Okay so then for consistency we should probably do the same thing on both Fedora and Ubuntu (install crun & runc).  Once the images are available, I'll open PRs for podman and buildah and some of the other repos.  Both to be on the safe-side, and because many/most projects are using old-ish images.

File: cache_images/fedora_packaging.sh
Comment (edsantiago) [inline cache_images/fedora_packaging.sh:148]: Could I kindly ask for a maintainer-friendly comment here? Something like "we download the rpm for these packages, but don't install them; tests wishing to use them can install them by running dnf install /path/to/where/the/rpm/was/downloaded"

File: cache_images/ubuntu_packaging.sh
Comment (edsantiago) [inline cache_images/ubuntu_packaging.sh:122]: Like the above, but in ubuntu-speak ("apt", etc)

**Review by edsantiago — APPROVED** [review]
LGTM, other than two future-friendly comment requests.

I spent many minutes considering whether the `REMOVE_PACKAGES` code should be left in (something I rarely do for dead code!) but no, I agree with your decision to excise it. Let's trust that we never need anything like that again.

File: cache_images/fedora_packaging.sh
Comment (cevich) [reply cache_images/fedora_packaging.sh:148]: I surely hope all possible future maintainers know `package==rpm` and use `dnf` to install them (similar for Ubuntu) :rofl:  I've got a ***bunch** of testing to do with these images once they build, so I'd prefer to prioritize that over a comment verbosity fix.  Though if I need to re-roll the images again here for some reason, I'm happy to copy-paste your suggestion in.

Comment (edsantiago) [reply cache_images/fedora_packaging.sh:148]: What I meant was, it is not clear where the downloads go. But OK, a careful reader can figure it out. Once you get that flake fixed, feel free to merge.

Comment (cevich) [reply cache_images/fedora_packaging.sh:148]: You're in luck, something else broke.  Adding your suggested comment...

**Comment by cevich** [comment]
Testing new images: b6110627968057344 and c6110627968057344

**Comment by cevich** [comment]
Okay, this seems to be okay (having both runc & crun) in the images.  There were a few hiccups on some repos but they don't appear to be caused by the change.

**Pull request merged**
